knowledge-base/42Crunch/api-security-audit-action/action-security.yml

```diff
@@ -0,0 +1,15 @@
+name: '42Crunch REST API Static Security Testing'
+github-token:
+  action-input:
+    input: github-token
+    is-default: true
+  permissions: # NOTE: github_api call is made to upload results to Github Code Scanning
```

Comment:
@Devils-Knight in this case you will need to add a conditional permission. 
https://github.com/42Crunch/api-security-audit-action/blob/0f30ed58af51f0a13efe2213230853072462f56f/action.yaml#L28
In this case, it will need permission to upload results if a certain input is set to true. 

Please see this example for permissions needed for `code scanning`. 
https://github.com/step-security/secure-workflows/blob/cad27a5b6c722a7156cb5a6ee63040fb0453a36e/knowledge-base/github/codeql-action/upload-sarif/action-security.yml#L7

This is an example for condition where permission should be granted only when input exists and its value is set to a specified thing. 
https://github.com/step-security/secure-workflows/blob/cad27a5b6c722a7156cb5a6ee63040fb0453a36e/knowledge-base/aws-actions/configure-aws-credentials/action-security.yml#L9

You will need to combine the info from these two examples. Please let me know if you have questions.

Reply:
@varunsh-coder sir for the above case we can simply check ` ${{ contains(with, upload-to-code-scanning)}}` but :
```yml
uses: 42Crunch/api-security-audit-action@v2
with:
  api-token: ${{ secrets.API_TOKEN }} 
  upload-to-code-scanning: false
```
for this example it would fail.
so my <b>question</b> is that **_how can we access the value associated within a_ `with attribute`**?

And also you have used `contains` which take 2 parameters 1st is array and 2nd the value to be found. so how is all the with values going inside `with` named array.
Is it because of this piece of code.
https://github.com/step-security/secure-workflows/blob/488698ce45c707445660ee5bfd96656fe8c1a6da/permissions.go#L234
so to answer my question do I have to update
https://github.com/step-security/secure-workflows/blob/488698ce45c707445660ee5bfd96656fe8c1a6da/permissions.go#L245
with something like this:
```go
gval.Function("contains", func(args ...interface{}) (interface{}, error) {
			switch v := args[0].(type) {
			case With:
				inputMap := v
				key := args[1].(string)
				val , found := inputMap[key]
                                if args[2] != nil {
                                      found := found && (val == args[2].(string))
                                }
				return (bool)(found), nil
			}
			return nil, fmt.Errorf("type not supported %T", args[0])
		}
```
And then we can use `${{ contains(with, upload-to-code-scanning, true)}}` to compare value also.

Reply:
@Devils-Knight In this case, you will need to use both contains and also check for the value. 

Please see this example:
https://github.com/step-security/secure-workflows/blob/746fc967be57105418e1e434f8baa0d4ebae491c/knowledge-base/reviewdog/action-golangci-lint/action-security.yml#L9

I had forgotten to tell you about how to check the value, sorry about that. 

So, here it would be something like:
```
security-events-if: ${{ contains(with, 'upload-to-code-scanning') && with['upload-to-code-scanning'] == 'true'
```